test/es-module/test-typescript-commonjs.mjs

```diff
@@ -59,12 +59,23 @@ test('require a .ts file with implicit extension fails', async () => {
 });
 
 test('expect failure of an .mts file with CommonJS syntax', async () => {
-  const result = await spawnPromisified(process.execPath, [
-    fixtures.path('typescript/cts/test-cts-but-module-syntax.cts'),
-  ]);
+  const testFilePath = fixtures.path('typescript/cts/test-cts-but-module-syntax.cts');
+  const result = await spawnPromisified(process.execPath, [testFilePath]);
+
+  assert.strictEqual(result.stdout, '');
+
+  const expectedWarning = `Failed to load the ES module: ${testFilePath}. Make sure to set "type": "module" in the nearest package.json file or use the .mjs extension.`;
+  try {
+    assert.ok(result.stderr.includes(expectedWarning));
```

Comment:
I think you are basically replicating `assert.match(result.stderr, /Failed to load the ES module:.*test-cts-but-module-syntax\.cts ......../)` here?

Reply:
I edited, thanks for suggestion

Reply:
Hi @joyeecheung ,

Thanks for your suggestion about using assert.ok(result.stderr.includes(expectedWarning)). I tried replacing the regex-based assertion with that approach, but I'm now getting this new error:

```
Test failure: 'expect failure of an .mts file with CommonJS syntax'
AssertionError [ERR_ASSERTION]: The input did not match the expected warning…
```
It seems like the actual output still doesn’t match the string in expectedWarning. Do you have any idea why this might be happening or how I could fix it?

Thanks in advance for your help!

Reply:
I wonder if you have any suggestions @jasnell